jobs/build-and-deploy-publish-worker.groovy

```diff
@@ -0,0 +1,107 @@
+
+new Setup(steps
+
+).addStringParam(
+    "GIT_REVISION",
+    """The name of a webapp branch to use when building the docker image.
+Most of the time master (the default) is the correct choice. But it can
+be any commit that is "later" than the one for the last publish worker.""",
+    "master"
+
+).addStringParam(
+    "ZND_NAME",
+    """If you specified the commit of a znd in GIT_REVISION, put the
+znd's name here.  It will be appended to the docker image name, to
+indicate that this image does not derive from prod.""",
+    ""
+
+).addBooleanParam(
+    "VALIDATE_COMMIT",
+    """If set, check that this commit is safe to publish.  <b>UNSET
+WITH CAUTION!</b> -- only for situations where you are overwriting
+someone else's publish-deploy on purpose.""",
+    true
+
+).addBooleanParam(
+    "DEPLOY_TO_PROD",
+    """If set, actually deploy the new publish-worker image to production. If not set, the job will stop after building the image.""",
+    false
+
+).addStringParam(
+    "DEPLOYER_EMAIL",
+    """The email address of the deployer. This will be used for deployment tracking and auditing purposes.""",
+    ""
+	
+).apply();
+
+currentBuild.displayName = "${currentBuild.displayName} (${params.GIT_REVISION})";
+
+def _setupWebapp() {
+   withTimeout('1h') {
+   		kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", 
+			params.GIT_REVISION);
+	}
+}
+
+def runScript() {
+   // Prune docker images before building if under 1.5GB of disk space.
+   sh('[ $(df -BM --output=avail . | tr -cd 0-9) -gt 1500 ] || docker image prune -af');
+
+   dir("webapp") {
+       if (params.VALIDATE_COMMIT) {
+          sh("services/content-editing/publish/tools/validate_commit_for_publish.sh");
+       }
+       def buildOutput = exec.outputOf(["env", "ZND_NAME=${params.ZND_NAME}", "services/content-editing/publish/tools/build_publish_image.sh"]);
+       echo("For how to use this image, see");
+       echo("https://khanacademy.atlassian.net/wiki/spaces/CP/pages/299204611/Publish+Process+Technical+Documentation");
+
+       if (!params.DEPLOY_TO_PROD) {
+           echo("DEPLOY_TO_PROD is not set. Skipping deployment to production.")
+           return
+       }	   
+       def matcher = (buildOutput =~ /export PUBLISH_VERSION=([\w.-]+)/)
+       def publishVersion = matcher ? matcher[0][1] : null
+       matcher = null
+       if (!publishVersion) {
+           notify.fail("Could not extract PUBLISH_VERSION from build_publish_image.sh output")
+       }	   
+
+       def deployerEmail = params.DEPLOYER_EMAIL
+       if (!deployerEmail || !deployerEmail.endsWith("@khanacademy.org")) {
+           notify.fail("Could not determine a valid deployer email. DEPLOYER_EMAIL is missing or is a @khanacademy.org address: ${deployerEmail}")
+       }
+       def goArgs = [
+           "go", "run", "services/content-editing/cmd/update-publish-worker/main.go",
+           "--publish-image-version=${publishVersion}",
+           "--publish-image-version-including-master",
```

Comment:
I think this option in the script is not useful: https://github.com/Khan/webapp/blob/0656aa2d1665941d33bacc99d3f539421434dacd/services/content-editing/cmd/update-publish-worker/main.go#L97-L102

It's in conflict if `GIT_VERSION` is something other than "master", and `VALIDATE_COMMIT` is checked off.

I think we should drop it from `update-publish-worker/main.go`.